twilight-model/src/guild/emoji.rs

```diff
@@ -10,21 +10,18 @@ use serde::{Deserialize, Serialize};
 #[allow(clippy::struct_excessive_bools)]
 #[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
 pub struct Emoji {
-    #[serde(default)]
-    pub animated: bool,
-    #[serde(default)]
-    pub available: bool,
-    // This does not need to be optional here as it can only be optional
-    // in a unicode emoji. Which can only happen in reactions, and we use
-    // another struct for emojis in that case.
-    pub id: Id<EmojiMarker>,
-    #[serde(default)]
-    pub managed: bool,
-    pub name: String,
-    #[serde(default)]
-    pub require_colons: bool,
-    #[serde(default, skip_serializing_if = "Vec::is_empty")]
-    pub roles: Vec<Id<RoleMarker>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub animated: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub available: Option<bool>,
+    pub id: Option<Id<EmojiMarker>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub managed: Option<bool>,
+    pub name: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub require_colons: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub roles: Option<Vec<Id<RoleMarker>>>,
```

Comment:
idk about this change. I'm not really convinced on that. Need to think about a better approach on that one.

Reply:
Discord documents it as optional/nullable so I think we should reflect on that.